tests/kafkatest/tests/core/partitioned_produce_consume_test.py

```diff
@@ -0,0 +1,95 @@
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.tests.produce_consume_validate import ProduceConsumeValidateTest
+from kafkatest.utils import is_int_with_prefix
+
+
+class PartitionedProduceConsumeTest(ProduceConsumeValidateTest):
+    """
+    Validate that we can continue to produce and consume in the presence of a network partition.
+    """
+
+    def __init__(self, test_context):
+        """:type test_context: ducktape.tests.test.TestContext"""
+        super(PartitionedProduceConsumeTest, self).__init__(test_context=test_context)
+
+        self.topic = "test_topic"
+        self.zk = ZookeeperService(test_context, num_nodes=3)
+        self.kafka = KafkaService(test_context, num_nodes=3, zk=self.zk, topics={self.topic:{
+                                                                    "partitions": 10,
+                                                                    "replication-factor": 3,
+                                                                    "configs": {"min.insync.replicas": 3}}})
+        self.trogdor = TrogdorService(context=test_context, agent_nodes=self.kafka.nodes)
+        self.num_partitions = 10
+        self.timeout_sec = 60
+        self.producer_throughput = 1000
+        self.num_producers = 1
+        self.messages_per_producer = 1000
+        self.num_consumers = 1
+        self.fractured = False
+
+    def setUp(self):
+        self.zk.start()
+
+    def teardown(self):
+        super(PartitionedProduceConsumeTest, self).teardown()
+        self.trogdor.stop()
+
+    def min_cluster_size(self):
+        return 9
+
+    @cluster(num_nodes=9)
+    def test_produce_consume(self):
+        self.logger.info("Starting test PartitionedProduceConsumeTest.")
+        self.trogdor.start()
+        self.kafka.security_protocol = "PLAINTEXT"
+        self.kafka.interbroker_security_protocol = self.kafka.security_protocol
+        self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka,
+                                           self.topic, throughput=self.producer_throughput,
+                                           message_validator=is_int_with_prefix, acks=-1, enable_idempotence=True)
+        self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka, self.topic,
+                                        consumer_timeout_ms=60000,
+                                        message_validator=is_int_with_prefix)
+        self.kafka.start()
+
+        self.run_produce_consume_validate(lambda: wait_until(
+            lambda: self.produce_consume_validate_complete(),
+            timeout_sec=120, backoff_sec=1,
+            err_msg="Producer did not produce all messages in reasonable amount of time"))
+        util.wait_until(self.node0_is_partitioned, 120,
+                        err_msg="Failed to verify that node0 was partitioned.")
+
+    def node0_is_partitioned(self):
+        self.kafka.nodes[0].account.ssh("nc -")
```

Comment:
I tried running this test as a docker test and got the exception:
```
  File "/opt/kafka-dev/tests/kafkatest/tests/core/partitioned_produce_consume_test.py", line 83, in test_produce_consume
    err_msg="Failed to verify that node0 was partitioned.")
  File "/usr/local/lib/python2.7/dist-packages/ducktape/utils/util.py", line 31, in wait_until
    if condition():
  File "/opt/kafka-dev/tests/kafkatest/tests/core/partitioned_produce_consume_test.py", line 86, in node0_is_partitioned
    self.kafka.nodes[0].account.ssh("nc -")
  File "/usr/local/lib/python2.7/dist-packages/ducktape/cluster/remoteaccount.py", line 251, in ssh
    raise RemoteCommandError(self, cmd, exit_status, stderr.read())
RemoteCommandError: ducker@ducker05: Command 'nc -' returned non-zero exit status 1. Remote error message: -: forward host lookup failed: Unknown host
```

Reply:
Sorry, that was a typo.  This should be using the same netcat invocation as the other test.  Fixed.